pulsar-client-api/src/main/java/org/apache/pulsar/client/api/Message.java

```diff
@@ -245,4 +245,20 @@
      * @since 2.8.0
      */
     void release();
+
+    /**
+     * Get broker publish time from broker entry metadata.
+     *
+     * @since 2.9.0
+     * @return broker publish time from broker entry metadata
+     */
+    long getBrokerPublishTime();
+
+    /**
+     * Get index from broker entry metadata.
+     *
+     * @Since 2.9.0
+     * @return index from broker entry metadata
+     */
+    long getIndex();
```

Comment:
The same here.
Also, can you briefly explain what is the meaning of this index?

Reply:
Thanks. The index is from [PR-9039](https://github.com/apache/pulsar/pull/11553), which provides continuous message sequence-Id for messages in one topic-partition.